assets/components/marketingConsent/marketingConsent.jsx

```diff
@@ -51,6 +62,33 @@ const MarketingConsent = (props: PropTypes): React.Node => {
   return content;
 };
 
+function mapDispatchToProps(dispatch: Dispatch<Action>) {
+  return {
+    onClick: (marketingPreferencesOptIn: boolean, email: string, csrf: CsrfState, context: string) => {
+      trackComponentClick('marketing-permissions-old-flow');
+      sendMarketingPreferencesToIdentity(
+        marketingPreferencesOptIn,
+        email,
+        dispatch,
+        csrf,
+        context,
+      );
+    },
+    marketingPreferenceUpdate: (preference: boolean) => {
+      dispatch(setGnmMarketing(preference));
+    },
+  };
+}
+
+function mapStateToProps(state) {
```

Comment:
Unfortunately shared components can't be `connect`ed to the redux store because they don't know the structure of the state on the page in which they're used. One manifestation of this problem is that there's no way to give this `state` argument a type annotation.

To give a shared component access to data in the redux store you can create a "container" component at the page level, which handles the `connect`ion.

Reply:
At this point I'm losing the will to live. But I'll have a look anyway.